test/user.model.test.js

```diff
@@ -84,6 +85,37 @@ describe('user.models.test.js', () => {
     });
   });
 
+  describe('#createUserWithCollective', () => {
+    it('uses "anonymous" slug if name is not provided', () => {
+      const userParams = { email: 'frank@zappa.com' };
+      return User.createUserWithCollective(userParams).then(user => {
+        expect(user.collective.slug.startsWith('anonymous')).to.equal(true);
+      });
+    });
+
+    it('uses "user" slug if name is not sluggifyable', () => {
+      return User.createUserWithCollective({
+        email: randEmail('user@domain.com'),
+        firstName: '...',
+        lastName: '?????',
+      }).then(user => {
+        expect(user.collective.slug.startsWith('user')).to.equal(true);
+      });
+    });
+
+    it('knows how to deal with special characters', () => {
+      return User.createUserWithCollective({
+        email: randEmail('user@domain.com'),
+        firstName: '很棒的用户',
+        lastName: 'awesome',
+      }).then(user => {
+        expect(user.collective.slug).to.equal(
+          'hen3-bang4-de-yong4-hu4-awesome',
```

Comment:
why do we have numbers in there?

Reply:
Apparently it is the standard way to represent these (mandarin) characters with Latin characters. Example: https://en.wiktionary.org/wiki/hen3